src/realm/sync/client.cpp

```diff
@@ -266,7 +279,8 @@ class SessionWrapper final : public util::AtomicRefCountBase, DB::CommitListener
     void on_flx_sync_error(int64_t version, std::string_view err_msg);
     void on_flx_sync_version_complete(int64_t version);
 
-    void report_progress(bool only_if_new_uploadable_data = false);
+    void init_progress_handler();
+    void report_progress(bool is_download, bool only_if_new_uploadable_data = false);
```

Comment:
could add a comment that `only_if_new_uploadable_data` can be true only if `is_download` is false

Reply:
added the comment and the check